tests/e2e-pw/utils/merchant-navigation.ts

```diff
@@ -94,3 +94,17 @@ export const goToConnect = async ( page: Page ) => {
 	);
 	await dataHasLoaded( page );
 };
+
+export const goToSubscriptionPage = async (
+	page: Page,
+	subscriptionId: number
+) => {
+	await page.goto( '/wp-admin/admin.php?page=wc-orders--shop_subscription', {
+		waitUntil: 'load',
+	} );
+	const orderRow = page.locator(
+		'tr#order-' + subscriptionId + ' .order_title a:nth-child(1)'
+	);
+	orderRow.evaluate( ( el: HTMLLinkElement ) => el.click() );
+	await dataHasLoaded( page );
```

Comment:
Here we can use the role link instead. We don't need `dataHasLoaded` here as it's only useful in WooPayments pages.

And I think we can rename this to be clearer, we're actually expecting the order ID.

```suggestion
	await page.getByRole( 'link', { name: `#${ subscriptionId }` } ).click()
```

Reply:
Sorry, I don't get it. Subscription ID and the Order ID are different numbers, and the order ID isn't being listed on the Subscriptions page.

Reply:
And if you directly go to the subscription page by URL, you may get a "page doesn't exist" screen. That's why I first went to the subscription listing page, then navigated through it.

Reply:
My bad, I overlooked it and thought that you were reading the `orderId` instead of the `subscriptionId` from the shopper order page.

Why are you getting the `page doesn't exist` error? 🤔 I think going directly to the subscription page should be the way to go.

Reply:
It goes too fast before the subscription record is created I guess, I don't know.

Reply:
That's weird, how we're getting the ID if it doesn't exist yet? 🤔 I think we should always try to simplify E2E tests and get rid of any extra steps, although this one doesn't lead to flakiness.

I'd leave it to you to decide, but you can still change the selector to use getByRole instead of locator 🙂